api/src/main/java/io/druid/timeline/TaskDataSegment.java

```diff
@@ -0,0 +1,94 @@
+package io.druid.timeline;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.joda.time.Interval;
+
+import java.util.Map;
+
+/**
+ * Created by haoxiang on 16/11/11.
+ */
+public class TaskDataSegment
+{
+
+    private final String type;
+    private final String id;
+    private final String groupId;
+    private final String dataSource;
+    private final Map<String, Object> ioConfig;
+    private final Interval interval;
+
+    @JsonCreator
+    public TaskDataSegment(
```

Comment:
Are we sure this class is needed? What is the class used for serde when writing to pending Segments table?